lib/uplink/ext/common.go

```diff
@@ -29,9 +40,128 @@ func GetIDVersion(number C.uint, cErr **C.char) C.struct_IDVersion {
 		return C.struct_IDVersion{}
 	}
 
-	if err := lib.GoToCStruct(goIDVersion, cIDVersion); err != nil {
+	if err := GoToCStruct(goIDVersion, cIDVersion); err != nil {
 		*cErr = C.CString(err.Error())
 		return cIDVersion
 	}
 	return cIDVersion
 }
+
+func GoToCStruct(fromVar, toPtr interface{}) error {
```

Comment:
Why do we need to user reflection for this? We shouldn't have too many structs needing the mapping?
There might be also simpler implementations for this.